### PR TITLE
Corrected minitests

### PR DESCRIPTION
**Pull request by rkorlepa**
Fixed the minitest case scenarios for failing if F3 card was not present in the switch.
Moved compatible_interface? to ciscotest.rb and included it where ever necessary.

This also contains a small fix related to getting all the instances of bridge-domain, it was causing issues in puppet layer so storing all the hash keys in string format rather than in integer.

### CODE REVIEW

File: lib/cisco_node_utils/bridge_domain.rb [hunk 1]
```diff
@@ -93,7 +93,7 @@ def self.bds
           bd_list.map { |elem| BridgeDomain.bd_ids_to_array(elem) }
           .flatten.uniq.sort
       final_bd_list.each do |id|
-        hash[id] = BridgeDomain.new(id, false)
+        hash[id.to_s] = BridgeDomain.new(id, false)
```
Comment (mikewiebe) [inline lib/cisco_node_utils/bridge_domain.rb:96]: Why is this change needed?

File: tests/test_bridge_domain.rb [hunk 2]
```diff
@@ -43,11 +43,11 @@ def teardown
   def test_single_bd_create_destroy
     bd = BridgeDomain.new('100', true)
     bds = BridgeDomain.bds
-    assert(bds.key?(100), 'Error: failed to create bridge-domain 100')
+    assert(bds.key?(100.to_s), 'Error: failed to create bridge-domain 100')
```
Comment (mikewiebe) [inline tests/test_bridge_domain.rb]: Why is this change needed?

File: tests/test_encapsulation.rb [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 require_relative 'ciscotest'
 require_relative '../lib/cisco_node_utils/encapsulation'
+require_relative '../lib/cisco_node_utils/vdc'
```
Comment (mikewiebe) [inline tests/test_encapsulation.rb:17]: Do you need to enable any features for these tests to run properly?

File: tests/ciscotest.rb [hunk 4]
```diff
@@ -204,4 +204,17 @@ def get_interface_cleanup_config(intf_name)
       ["default interface #{intf_name}"]
     end
   end
+
+  def compatible_interface?
```
Comment (mikewiebe) [inline tests/ciscotest.rb]: Please change this method name to `mt_full_interface?`.  This is very specific to mtfull and now that it's not in a standalone file, we need to be more specific with the name.

File: tests/test_encapsulation.rb (hunk 3 above)
Comment (rkorlepa) [inline tests/test_encapsulation.rb:17]: Was testing something and forgot to remove it..
Removed it now.

File: tests/ciscotest.rb (hunk 4 above)
Comment (rkorlepa) [inline tests/ciscotest.rb]: Changed it as suggested.

File: tests/test_bridge_domain.rb (hunk 2 above)
Comment (rkorlepa) [inline tests/test_bridge_domain.rb]: changed it to string instead of .to_s:
assert(bds.key?('100')

File: lib/cisco_node_utils/bridge_domain.rb (hunk 1 above)
Comment (rkorlepa) [inline lib/cisco_node_utils/bridge_domain.rb:96]: Wanted to keep all the return values consistent with other getters in this node util class, all the getter return string or bolean values.

File: tests/test_bridge_domain.rb [hunk 5]
```diff
@@ -43,11 +43,11 @@ def teardown
...
 
     bd.destroy
     bds = BridgeDomain.bds
-    refute(bds.key?(100), 'Error: failed to destroy bridge-domain 100')
+    refute(bds.key?(100.to_s), 'Error: failed to destroy bridge-domain 100')
```
Comment (mikewiebe) [inline tests/test_bridge_domain.rb]: Did you mean to change these to `100` instead of `100.to_s`?

Comment (rkorlepa) [inline tests/test_bridge_domain.rb]: Yes forgot to push the changes, have done it now.
